include/seqan3/core/algorithm/parameter_pack.hpp

```diff
@@ -0,0 +1,37 @@
+
+#pragma once
+
+#include <utility>
+#include <seqan3/std/concepts>
+
+namespace seqan3::detail
+{
+
+/*!\brief Applies a function to a range of arguments.
+ * \ingroup algorithm
+ *
+ * \tparam unary_function_t The function type, like function pointers, functors and lambdas.
+ * \tparam ...args_t        The parameter pack of the arguments. (Note that each argument type can be different)
+ *
+ * \param  fn               The function to call on every argument.
+ * \param  args             The arguments as parameter pack.
+ *
+ * \details This function behaves like std::for_each but on parameter packs.
+ * The invocation will be done without any loop.
+ *
+ * \include test/snippet/core/algorithm/parameter_pack.cpp
+ */
+template <typename unary_function_t, typename ...args_t>
+//!\cond
+    requires (std::Invocable<unary_function_t, args_t> && ... && true)
+//!\endcond
+constexpr void parameter_pack_for_each(unary_function_t && fn, args_t && ...args)
```

Comment:
First we already use it and second we can replace some of the meta:: library calls if we introduce this. I will add a `for_each_type` metafunction to this PR. I need this for example in a test case where I generate a lot of types for a base type and then test all generated types via a given function. This will be needed in a coming PR for SIMDs.